packages/@aws-cdk/aws-iam/lib/private/imported-role.ts

```diff
@@ -0,0 +1,88 @@
+import { Resource, Token, TokenComparison, Annotations } from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { Grant } from '../grant';
+import { IManagedPolicy } from '../managed-policy';
+import { Policy } from '../policy';
+import { PolicyStatement } from '../policy-statement';
+import { IComparablePrincipal, IPrincipal, ArnPrincipal, AddToPrincipalPolicyResult, PrincipalPolicyFragment } from '../principals';
+import { IRole, FromRoleArnOptions } from '../role';
+import { AttachedPolicies } from '../util';
+
+export interface ImportedRoleProps extends FromRoleArnOptions {
+  readonly roleArn: string;
+  readonly roleName: string;
+  readonly account?: string;
+}
+
+export class ImportedRole extends Resource implements IRole, IComparablePrincipal {
```

Comment:
Any overlap worth exploring with 

```
    class Import extends Resource implements IRole, IComparablePrincipal {
      public readonly grantPrincipal: IPrincipal = this;
      public readonly principalAccount = roleAccount;
      public readonly assumeRoleAction: string = 'sts:AssumeRole';
      public readonly policyFragment = new ArnPrincipal(roleArn).policyFragment;
      public readonly roleArn = roleArn;
      public readonly roleName = roleName;
      private readonly attachedPolicies = new AttachedPolicies();
      private readonly defaultPolicyName?: string;
      private defaultPolicy?: Policy;
```
?

Reply:
it should be exactly the same. I did not intend to change any functionality, so if anything just double check that I didn't change any functionality?

Reply:
I think I got some wires crossed here: If a class already exists with a full overlap of fields and methods, why does this class need to be defined?

And if (few) methods need different implementations, would an override not do the trick?